src/utils.ts

```diff
@@ -0,0 +1,85 @@
+export interface ImageInfo {
+    type: string;
+    width: number;
+    height: number;
+    image: HTMLImageElement;
+}
+
+function loadImageInfo(url: string): Promise<ImageInfo> {
+    const result = {
+        image: new Image(),
+        type: "",
+        width: 0,
+        height: 0
+    };
+
+    return fetch(url)
```

Comment:
It would be cool to be able to pass in fetch options e.g. for custom headers.

Reply:
Yes